### ISSUE TITLE
A socket operation error when running the Lambda test tool

### ISSUE THREAD

**Issue by Aliraza-N** [aws/aws-toolkit-visual-studio#213]
I am trying to test a Lambda function using the test tool, and I'm seeing this error (attached). 
[err.txt](https://github.com/aws/aws-toolkit-visual-studio/files/7952927/err.txt)

I'm guessing it's a credential issue, however I am able to see objects in my AWS account in Visual studio using the toolkit:
![image](https://user-images.githubusercontent.com/24211346/151420661-0f4a121a-230c-4061-87dc-c3f4aadac35b.png)

For reference I'm using SSO credentials.

**Comment by awschristou** [comment]
Hi @Aliraza-N , are you getting this error when you try to press the "Execute Function" button on the test tool, or is the test tool able to successfully start debugging your code, and you get this error when you run a line of code where an AWS Service Client is making an API call? Also, are you able to provide a small repro case?

**Comment by Aliraza-N** [comment]
Hi, thanks for following up. I was actually able to solve this issue, but for the sake of future queries:
The issue occurred when running a line in my code where we made a call to an AWS service. Yes, it was due to improperly setting up credentials. Specifically, I had to set the environment variable "AWS_PROFILE" inside of `aws-lambda-tools-defaults.json`. This problem started occurring after switching to SSO-based authentication.
